compiler/damlc/tests/src/DA/Test/ScriptService.hs

```diff
@@ -43,17 +44,25 @@ import Test.Tasty
 import Test.Tasty.HUnit
 import Text.Regex.TDFA
 
-lfVersion :: LF.Version
-lfVersion = LF.versionDefault
-
 main :: IO ()
 main =
+    defaultMain $
+        testGroup
+            "Script Service"
+            [ withResourceCps (withScriptService lfVersion) (testScriptService lfVersion)
+            | lfVersion <- map LF.defaultOrLatestStable [minBound @LF.MajorVersion .. maxBound]
+            ]
+
+withScriptService :: LF.Version -> (SS.Handle -> IO ()) -> IO ()
+withScriptService lfVersion action =
   withTempDir $ \dir -> do
     withCurrentDirectory dir $ do
       setEnv "TASTY_NUM_THREADS" "1" True
```

Comment:
this needs to happen outside `defaultMain`, otherwise tasty runs the tests in parallel and you get funky errors :)

Reply:
Thanks! Done.